internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+		}
+	}
+
+	// Restrict Scopes Validation
+	if len(model.RestrictedScopes.Elements()) > 0 && !model.RestrictScopes.ValueBool() {
+		resp.Diagnostics.AddError("restrict_scopes must be set to true to configure restricted_scopes.", "")
+	}
+
+	// OIDC Policy Validation
+	if internaltypes.IsDefined(model.OidcPolicy) {
+		oidcPolicy := model.OidcPolicy.Attributes()
+		pairwiseIdentifierUserType := oidcPolicy["pairwise_identifier_user_type"]
+		oidcPolicySectorIdentifierUri := oidcPolicy["sector_identifier_uri"]
+		if (pairwiseIdentifierUserType != nil && !pairwiseIdentifierUserType.(types.Bool).ValueBool()) && internaltypes.IsDefined(oidcPolicySectorIdentifierUri) {
+			resp.Diagnostics.AddError("sector_identifier_uri can only be configured when pairwise_identifier_user_type is set to true.", "")
+		}
+	}
+
+	// JWKS Settings Validation
+	if !internaltypes.IsDefined(model.JwksSettings) {
+		if internaltypes.IsDefined(model.TokenIntrospectionEncryptionAlgorithm) {
+			resp.Diagnostics.AddError("token_introspection_encryption_algorithm must not be configured jwks_settings is configured.", "")
+		}
+		if model.RequireSignedRequests.ValueBool() {
+			resp.Diagnostics.AddError("require_signed_requests must be false when jwks_settings is not configured.", "")
+		}
+	}
+}
+
+func readOauthClientResponse(ctx context.Context, r *client.Client, plan, state *oauthClientResourceModel) diag.Diagnostics {
+	var diags, respDiags diag.Diagnostics
+	state.Id = types.StringValue(r.ClientId)
+	state.ClientId = types.StringValue(r.ClientId)
+	state.Enabled = types.BoolPointerValue(r.Enabled)
+	state.RedirectUris = internaltypes.GetStringSet(r.RedirectUris)
+	state.GrantTypes = internaltypes.GetStringSet(r.GrantTypes)
+	state.Name = types.StringValue(r.Name)
+	state.Description = types.StringPointerValue(r.Description)
+	state.ModificationDate = types.StringValue(r.ModificationDate.Format(time.RFC3339Nano))
+	state.CreationDate = types.StringValue(r.CreationDate.Format(time.RFC3339Nano))
+	state.LogoUrl = types.StringPointerValue(r.LogoUrl)
+	state.DefaultAccessTokenManagerRef, respDiags = resourcelink.ToState(ctx, r.DefaultAccessTokenManagerRef)
+	diags.Append(respDiags...)
+	state.RestrictToDefaultAccessTokenManager = types.BoolPointerValue(r.RestrictToDefaultAccessTokenManager)
+	state.ValidateUsingAllEligibleAtms = types.BoolPointerValue(r.ValidateUsingAllEligibleAtms)
+	state.RefreshRolling = types.StringPointerValue(r.RefreshRolling)
+	state.RefreshTokenRollingIntervalType = types.StringPointerValue(r.RefreshTokenRollingIntervalType)
+	state.RefreshTokenRollingInterval = types.Int64PointerValue(r.RefreshTokenRollingInterval)
+	state.PersistentGrantExpirationType = types.StringPointerValue(r.PersistentGrantExpirationType)
+	state.PersistentGrantExpirationTime = types.Int64PointerValue(r.PersistentGrantExpirationTime)
+	if types.StringPointerValue(r.PersistentGrantExpirationTimeUnit).ValueString() == "" {
```

Comment:
Could this instead be a nil check? `if r.PersistentGrantExpirationTimeUnit == nil`?

Reply:
This value is default and isn't returned by PingFed if unchanged. I'll add a better check though.

Reply:
updated check to `if r.GetPersistentGrantExpirationTimeUnit() == ""`